design/baremetal-operator/node-specific-kernel-args.md

```diff
@@ -0,0 +1,166 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# pre-provisioning kernel argument support
+
+## Status
+
+planned
+
+## Summary
+
+This document proposes a new BareMetalHost spec attribute intended for
+configuring the pre-provisioning image's (usually IPA) kernel arguments.
+
+## Motivation
+
+There are situations where a user would need node specific kernel arguments
+for the pre-provisioning image. The reason could be the use of a mixed set of
+hardware, troubleshooting provisioning or boot issues, selectively configuring
+custom logic e.g. enabling IPA plugins or other services embedded in the
+pre-provisioning image.
+
+### Goals
+
+1. Implement a string array type optional spec field named
+   `preProvisionKernelArgs`
+2. Extend the generic `Provisioner` interface to support the new optional field
+3. Implement support for the `preProvisionKernelArgs` in the Ironic provisioner
+
```

Comment:
You also need to extend the PreprovisioningImage object to accept these params.

Reply:
That will create a wild inconsistency that will cause users file bugs for me with `preprovisioningExtraKernelParams` does not work :(

Reply:
Btw it's fair if you ask me to follow-up with PPI support myself if you don't have cycles for that. I just want it to be in the design.